### PR TITLE
Remove statefulSetUID parameter from GetWorkloadName.

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Remove statefulSetUID parameter from GetWorkloadName.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
We are not using statefulSetUID parameter in GetWorkloadName() method.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by mbobrovskyi** [comment]
> Looking at https://github.com/search?q=repo%3Akubernetes-sigs%2Fkueue%20GetWorkloadNameForOwnerWithGVK&type=code we pass the object UID to the function for other APIs. I think if the StatefulSet is recreated (different UID) we also want to recreate the same group and re-admit.

Yes, but it's empty every time :) Removed to avoid confusing other contributors.

**Comment by mimowo** [comment]
ok, but it is surprising that `ss.UID` is empty. I would expect every object, also StatefulSet to have an UID. Am I missing something?

**Comment by mbobrovskyi** [comment]
> ok, but it is surprising that `ss.UID` is empty. I would expect every object, also StatefulSet to have an UID. Am I missing something?

We can't get the UID in the Default function because it's executed before creation, so we don't have the UUID at that time.

File: pkg/controller/jobs/statefulset/statefulset_webhook.go [hunk 1]
```diff
@@ -140,6 +139,6 @@ func (wh *Webhook) ValidateDelete(context.Context, runtime.Object) (warnings adm
 	return nil, nil
 }
 
-func GetWorkloadName(statefulSetName string, statefulSetUID types.UID) string {
-	return jobframework.GetWorkloadNameForOwnerWithGVK(statefulSetName, statefulSetUID, gvk)
+func GetWorkloadName(statefulSetName string) string {
+	return jobframework.GetWorkloadNameForOwnerWithGVK(statefulSetName, "", gvk)
```
Comment (mimowo) [inline pkg/controller/jobs/statefulset/statefulset_webhook.go:143]: Please add a comment: "Passing empty UID as it is not available before object creation"

Comment (mbobrovskyi) [reply pkg/controller/jobs/statefulset/statefulset_webhook.go:144]: Good point! Thanks. Done.

**Review by mimowo — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**